docs/protocol-generator.md

```diff
@@ -72,30 +82,52 @@ The specific types which could be assigned to contents in a protocol specificati
+| Custom types<sup>1</sup>               | `<CT>`  | `ct:RegExp(^[A-Z][a-zA-Z0-9]*$)`                                                                              | `ct:DataModel`                           | Custom Class                       |
+| Primitive types            | `<PT>`  | `pt:bytes`                                                                                                    | `pt:bytes`                               | `bytes`                            |
+|                            |               | `pt:int`                                                                                                      | `pt:int`                                 | `int`                              |
+|                            |               | `pt:float`                                                                                                    | `pt:float`                               | `float`                            |
+|                            |               | `pt:bool`                                                                                                     | `pt:bool`                                | `bool`                             |
+|                            |               | `pt:str`                                                                                                      | `pt:str`                                 | `str`                              |
+| Primitive collection types | `<PCT>` | `pt:set[<PT>]`                                                                                          | `pt:set[pt:str]`                         | `FrozenSet[str]`                   |
+|                            |               | `pt:list[<PT>]`                                                                                         | `pt:list[pt:int]`                        | `Tuple[int, ...]`<sup>*</sup>      |
+| Primitive mapping types<sup>2</sup>    | `<PMT>` | `pt:dict[<PT>, <PT>]`                                                                 | `pt:dict[pt:str, pt:bool]`               | `Dict[str, bool]`                  |
+| Multi types                | `<MT>`  | `pt:union[<PT>/<CT>/<PCT>/<PMT>, ..., <PT>/<CT>/<PCT>/<PMT>]` | `pt:union[ct:DataModel, pt:set[pt:str]]` | `Union[DataModel, FrozenSet[str]]` |
+| Optional types             | `<O>`   | `pt:optional[<MT>/<PMT>/<PCT>/<PT>/<CT>]`                                       | `pt:optional[pt:bool]`                   | `Optional[bool]`                   |
 
 <sup>*</sup> <sub>This is how variable length tuples containing elements of the same type are declared in Python; see <a href="https://docs.python.org/3/library/typing.html#typing.Tuple" target=_blank>here</a> </sub>
 
+#### Notes
+
+1. Currently, there is no way to describe custom types in a language independent format. This means that if a protocol specification includes custom types, the required implementations must be provided manally.
```

Comment:
`there is no way to describe custom types in a target language independent format` ..or something like that, language independent does not make sense as the specification language is also a language..

Reply:
Done